packages/mdc-fab/mdc-fab.scss

```diff
@@ -12,98 +12,58 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-@import "@material/animation/functions";
-@import "@material/animation/variables";
-@import "@material/elevation/mixins";
-@import "@material/ripple/mixins";
-@import "@material/theme/mixins";
+@import "@material/theme/variables";
+@import "./mixins";
 
 // postcss-bem-linter: define fab
-.mdc-fab {
-  @include mdc-ripple-base;
-  @include mdc-ripple-bg((pseudo: "::before", base-color: white, opacity: .16));
-  @include mdc-ripple-fg((pseudo: "::after", base-color: white, opacity: .16));
-
-  display: inline-flex;
-  position: relative;
-  justify-content: center;
-  width: 56px;
-  height: 56px;
-  padding: 0;
-  transition: box-shadow 280ms $mdc-animation-standard-curve-timing-function;
-  border: none;
-  border-radius: 50%;
-  cursor: pointer;
-  user-select: none;
-  box-sizing: border-box;
-  fill: currentColor;
-  -moz-appearance: none;
-  -webkit-appearance: none;
-  overflow: hidden;
-
-  &:not(.mdc-ripple-upgraded) {
-    -webkit-tap-highlight-color: rgba(0, 0, 0, .18);
-  }
-
-  @include mdc-theme-prop(background-color, secondary);
-  @include mdc-theme-prop(color, text-primary-on-secondary);
-  @include mdc-elevation(6);
 
-  &--mini {
-    width: 40px;
-    height: 40px;
-  }
-
-  &--plain {
-    background-color: white;
-
-    @include mdc-theme-prop(color, text-primary-on-light);
-    @include mdc-ripple-bg((pseudo: "::before"));
-    @include mdc-ripple-fg((pseudo: "::after"));
-  }
-
-  &:active,
-  &:focus {
-    outline: none;
-  }
+$mdc-theme-property-values-custom: map-merge(
```

Comment:
I don't think we should be adding keys to this map for each component, but I realize this is here because we only want to look up parameterized colors via the map...

I realize this is currently here primarily for plain and disabled variants, both of which we're planning to remove for FAB (plain should be easily achievable via the theme mixin, and disabled FABs shouldn't really be a thing), though we will have to answer this question eventually as it relates to disabled buttons.

As for the tap highlight color, can we compute that based on the foreground color plus this opacity setting, so it doesn't need to be a parameter?